### DISCUSSION TITLE
1.22.0

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#9445] [General]
# Notes
- The TypeScript refactor has been finished therefore Zigbee2MQTT is now fully written in TypeScript. This refactor brings some changes to the external extensions. The `onMQTTConnected()` and `onZigbeeStarted()` are replaced with `start()`, the `onMQTTMessage()` and `onZigbeeEvent()` methods are not called automatically anymore and require subscription through the `eventbus` ([documentation](https://www.zigbee2mqtt.io/information/user_extensions.html)).
- With this release also comes the launch of the new site, created by [psi-4ward](https://github.com/psi-4ward). Make sure to check it out at [zigbee2mqtt.io](https://zigbee2mqtt.io)! 

# Features
- #6281 The availability feature has been reimplemented. Adding more features, improving reliability and putting less stress on the network. Old configurations will still work (`advanced.availability_timeout`) but it is recommended to update your `configuration.yaml`. ([documentation](https://www.zigbee2mqtt.io/information/availability.html))
- #9056 Allow to create and recall scenes in the frontend
![image](https://user-images.githubusercontent.com/2892853/136820642-a1d4f79e-528b-4c21-9dd7-bf2a0ad0a373.png)
- #9110 Allow to manage device options in frontend
<img width="696" alt="Screenshot 2021-10-23 at 10 53 51" src="https://user-images.githubusercontent.com/2892853/138549973-2aa6bea2-e2bb-407a-bc4b-c2f328b3e159.png">

# Improvements
- [#3109](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3109) Adds OTA support for Develco EMIZB-132
- [#3112](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3112) Support minimum_on_level for ubisys D1 / D1-R
- [#3121](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3121) Expose `battery_low` for NAS-PD07
- #8956 Stop Zigbee2MQTT with error when initial MQTT connect fails
- [#3119](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3119) Add firmware version reporting for Develco EMIZB-132 and SMSZB-120
- [#3083](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3083) Adding pin code programming, presenting action and auto relock for easyCodeTouch_v1
- #8287 Home Assistant: discover last_seen sensor when enabled
- [#3068](https://github.com/Koenkk/zigbee-herdsman-converters/issues/3068) Improve TuYa device interviewing stability
- [#926](https://github.com/nurikk/zigbee2mqtt-frontend/issues/926) Allow publishing to device by endpoint ID
- [#3158](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3158) Expose temperature for Xiaomi SJCGQ11LM
- [#3164](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3164) Improve support for Busch-Jaeger 6735/6736/6737
- [#3166](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3166) Support mode_switch for Xiaomi QBKG39LM, QBKG38LM and QBKG25LM
- [#9214](https://github.com/Koenkk/zigbee2mqtt/pull/9214) Expose device and group configuration_url to Home Assistant
- #9147 Woox R7049 support smoke test
- #9312 Add support for unit of measurement to HA number entities
- [#3227](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3227) Add operation mode for YSR-MINI-Z
- [#3222](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3222) Expose `illuminance_above_threshold` and add `illuminance_below_threshold_check` for IKEA E1525/E1745
- [#3241](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3241) Expose 2 endpoints for OSRAM 4062172044776

# Fixes
- [#3066](https://github.com/Koenkk/zigbee-herdsman-converters/issues/3066) Fix EMIZB-132 divisor
- #7423 Fix last seen not published when changed
- [#3107](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3107) Fix power monitoring for TuYa TS011F with manufacturer name `_TZ3000_dpo1ysak`
- #8959 Fix external converters not loading
- [#3145](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3145) Fix incorrect battery percentage for some Yale locks
- #8769 #8617 Fix Home Assistant template variable warnings
- #9085 Fix Node.js 10 compatibility
- [#9133](https://github.com/Koenkk/zigbee2mqtt/issues/9133) Fix crash when converter throws exception
- [#3171](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3171) Fix incorrect naming for window_open_internal (closing -> closed)
- [#3176](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3176) Fix `setpoint_change_source` for Danfoss and Popp thermostats
- [#3122](https://github.com/Koenkk/zigbee-herdsman-converters/issues/3122) Home Assistant fix discovery for covers only supporting tilt
- [#3086](https://github.com/Koenkk/zigbee-herdsman-converters/issues/3086) Fix button 2, 3 and 4 not working for Sunricher SR-ZG9001K12-DIM-Z4
- [#3202](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3203) Fix power source for 3RWS18BZ
- [#3201](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3201) Fix `running_state` always `null` for Eurotronic thermostats
- #9268 Fix Home Assistant discovery for select having numbers as options 
- [#3218](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3218) Fix TRADFRI LED1624G9 not supporting color temperature
- #9057 Fix power measurements for TS011F _TZ3000_cphmq0q7 wit appVersion 69
- [#3229](https://github.com/Koenkk/zigbee-herdsman-converters/issues/3229) Fix value_min/max not correct for Home Assistant number when 0
- #8349 Fix brightness command timing out for GL-SD-001

# New supported devices
This release adds support for 75 new devices:
- `3157100-E` Centralite 3-Series pearl touch thermostat,
- `322054` Lanesto Dimmable led driver
- `3261348P6` Philips Hue white ambiance Still
- `5060731P7` Philips Hue White & Color ambience Centris ceiling light (4 spots)
- `5060931P7` Philips Hue White & Color Ambiance Centris ceiling light (3 spots)
- `5110.40` Iluminize Zigbee 3.0 LED controller multi 5 - 4A,RGB W/CCT LED
- `5112.80` Iluminize Zigbee 3.0 LED-controller 1x 8A
- `5633031P9` Philips Hue White ambiance Pillar spotlight
- `6109231C5` Philips Hue white ambiance Apogee square
- `6109331C5` Philips Hue white ambiance Apogee round
- `70049` Bega Zigbee control module DALI
- `8719514301481` Philips Hue Filament Globe Ambiance E27
- `8719514301542` Philips Hue Filament Globe XL Ambiance E27
- `9134` LED Trading Powerstrip with 4 sockets and USB
- `9290018217` Philips Hue white A60 bulb B22 bluetooth
- `929002376301` Philips Hue Iris rose limited edition (generation 4) 
- `92900244777` Philips Hue White Ambiance E27 ST64 filament bulb
- `9290024684` Philips Hue white ambiance E27 1100lm with Bluetooth
- `929002469202` Philips Hue white A60 bulb E27 1050lm with Bluetooth
- `929002471601` Philips Hue white and color ambiance E26/E27 1600lm
- `929002477901` Philips Hue white filament Edison ST72 E27 LED warm-to-cool
- `929002478401` Philips Hue white filament Edison ST19 E26 LED warm-to-cool
- `929002966401` Philips Hue White & Color Ambiance Surimu square panel
- `929002966501` Philips Hue White and Color Ambiance Surimu rectangle panel
- `929003050601` Philips Hue smart plug
- `BSM-300ZB` ShinaSystem SiHAS remote control
- `CSM-300ZB` ShinaSystem SiHAS multipurpose sensor
- `DSM-300ZB` ShinaSystem SiHAS contact sensor
- `EFEKTA_ePWS` Custom devices (DiY) [Plant wattering sensor with e-ink display](https://efektalab.com/epws102)
- `HK-DIM-A` Candeo Zigbee LED dimmer smart switch
- `Icon` Danfoss Icon floor heating (regulator, Zigbee module & thermostats)
- `KEYZB-110` Develco Keypad
- `KP-23EL-ZBS-ACE` Climax Remote Keypad
- `LDSENK10` ADEO LEXMAN motion sensor
- `LED1949C5` IKEA TRADFRI LED bulb E14 470 lumen, wireless dimmable white spectrum/chandelier opal white
- `MIR-HE200-TY` TuYa Human presence sensor
- `MSM-300ZB` ShinaSystem SiHAS remote control 4 button
- `NCZ-3010` Nyce Door hinge sensor
- `PECLS01` Perenio Flood alarm device
- `PECWS01` Perenio Door sensor
- `QBKG19LM` Xiaomi Aqara smart wall switch T1 (with neutral, single rocker)
- `QBKG20LM` Xiaomi Aqara smart wall switch T1 (with neutral, double rocker)
- `QBKG34LM` Xiaomi Aqara smart wall switch T1 (with neutral, three rocker)
- `ROB_200-025-0` ROBB Zigbee 8 button wall switch
- `S24013` The Light Group SLC SmartOne AC dimmer mini 200W Zigbee LN
- `S57003` The Light Group SLC SmartOne Zigbee wall remote 4-channels
- `SBM300Z1` ShinaSystem SiHAS IOT smart switch 1 gang
- `SBM300Z2` ShinaSystem SiHAS IOT smart switch 2 gang
- `SBM300Z3` ShinaSystem SiHAS IOT smart switch 3 gang
- `SBM300Z4` ShinaSystem SiHAS IOT smart switch 4 gang
- `SBM300Z5` ShinaSystem SiHAS IOT smart switch 5 gang
- `SBM300Z6` ShinaSystem SiHAS IOT smart switch 6 gang
- `SBM300ZB1` ShinaSystem SiHAS remote control
- `SBM300ZB2` ShinaSystem SiHAS remote control 2 button
- `SBM300ZB3` ShinaSystem SiHAS remote control 3 button
- `SMART-HEAT10` Alecto Radiator valve with thermostat
- `TS0001_switch_module` TuYa 1 gang switch module
- `TS0002_switch_module` TuYa 2 gang switch module
- `TS0003_switch_module` TuYa 3 gang switch module
- `TS0004_switch_module` TuYa 4 gang switch module
- `TS0111_valve` TuYa Smart water/gas valve
- `TS011F_plug_1` TuYa Smart plug (with power monitoring)
- `TS011F_plug_3` TuYa Smart plug (with power monitoring by polling)
- `TS0601_motion_sensor` TuYa Human presence sensor AIR
- `TS0601_rcbo` TuYa DIN mount RCBO with smart energy metering
- `TS0601_thermostat_1` TuYa Thermostatic radiator valve
- `TS1101_dimmer_module` TuYa 2CH Zigbee dimmer module
- `TSM-300ZB` ShinaSystem SiHAS temperature/humidity sensor
- `TV01-ZB` Moes Thermostat radiator valve
- `USM-300ZB` ShinaSystem SiHAS multipurpose sensor
- `WDE002386` Schneider Electric Push button dimmer
- `WDE002906` Schneider Electric Wiser wireless switch 1-gang
- `YRL226 TS` Yale Assure lock SL
- `YSR-MINI-Z` TuYa 2 in 1 dimming remote control and scene control
- `ZTS-EU_4gang` Moes Wall touch light switch (4 gang)

<hr /><em>This discussion was created from the release <a href='https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.22.0'>1.22.0</a>.</em>

**Comment by jkpe** [comment]
The availability feature looks great. Should Home Assistant now see availability as an entity against a device?

**Comment by michal-bartak** [comment]
I can see "Empty exposes definition" in Settings (specific) tab.
![image](https://user-images.githubusercontent.com/7868445/140606997-ea16a658-8fe7-4856-8a95-5a7bddbc7d40.png)
Is it expected? Doesn't sound English
